src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -0,0 +1,155 @@
+    >('balances.providers.zerion.chains');
+  }
+
+  async getBalances(args: {
+    chainId: string;
+    safeAddress: string;
+    fiatCode: string;
+  }): Promise<Balance[]> {
+    try {
+      const cacheDir = CacheRouter.getZerionBalancesCacheDir(args);
+      const chainName = this._getChainName(args.chainId);
+      const currency = args.fiatCode.toLowerCase();
+      const url = `${this.baseUri}/v1/wallets/${args.safeAddress}/positions?filter[chain_ids]=${chainName}&currency=${currency}&sort=value`;
+      const { data } = await this.dataSource.get<ZerionBalances>({
+        cacheDir,
+        url,
+        notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
+        networkRequest: { headers: { Authorization: `Basic ${this.apiKey}` } },
+        expireTimeSeconds: this.defaultExpirationTimeInSeconds,
+      });
+      return this._mapBalances(chainName, data);
+    } catch (error) {
+      throw new DataSourceError(
+        `Error getting ${args.safeAddress} balances from provider: ${asError(error).message}}`,
+      );
+    }
+  }
+
+  private _mapBalances(
+    chainName: string,
+    zerionBalances: ZerionBalance[],
+  ): Balance[] {
+    return zerionBalances
+      .filter((zb) => zb.attributes.flags.displayable === true)
+      .map((zb) => {
+        const implementation = zb.attributes.fungible_info.implementations.find(
+          (implementation) => implementation.chain_id === chainName,
+        );
+        if (!implementation)
+          throw Error(
+            `Zerion error: ${chainName} implementation not found for balance ${zb.id}`,
+          );
+        const fiatBalance = getNumberString(zb.attributes.value ?? 0);
+        const fiatConversion = getNumberString(zb.attributes.price);
+
+        return {
+          ...(implementation.address === null
+            ? this._mapNativeBalance(zb.attributes)
+            : this._mapErc20Balance(zb.attributes, implementation.address)),
+          fiatBalance,
+          fiatConversion,
+        };
+      });
+  }
+
+  private _mapErc20Balance(
+    zerionBalanceAttributes: ZerionAttributes,
+    tokenAddress: string,
+  ): Erc20Balance {
+    const { fungible_info, quantity } = zerionBalanceAttributes;
+    return {
+      tokenAddress,
+      token: {
+        name: fungible_info.name!,
+        symbol: fungible_info.symbol!,
+        decimals: quantity.decimals,
+        logoUri: fungible_info.icon.url ?? '',
+      },
+      balance: quantity.int,
+    };
+  }
+
+  private _mapNativeBalance(
+    zerionBalanceAttributes: ZerionAttributes,
+  ): NativeBalance {
+    return {
+      tokenAddress: null,
+      token: null,
+      balance: zerionBalanceAttributes.quantity.int,
+    };
+  }
+
+  async clearBalances(args: {
+    chainId: string;
+    safeAddress: string;
+  }): Promise<void> {
+    const key = CacheRouter.getZerionBalancesCacheKey(args);
+    await this.cacheService.deleteByKey(key);
+  }
+
+  private _getChainName(chainId: string): string {
```

Comment:
Nit: this confused me again as in the Valk implementation (I was expecting "Polygon", for example). I think we should change this to be `shortName`, etc. in a follow up PR.

Reply:
Ok, makes sense to me, will do it in a separate PR then 👍🏻 thanks